src/components/spreadsheet/equipment-table.js

```diff
@@ -29,8 +35,74 @@ export const EquipmentTable = ({
     fetched,
     network,
     shouldHidePinnedHeaderRightBorder,
+    onCellClicked,
 }) => {
     const theme = useTheme();
+    const { snackError } = useSnackMessage();
+    const [clickedCellData, setClickedCellData] = React.useState({});
+    const [generatorId, setGeneratorId] = React.useState('');
+    const [
+        popupGeneratorEditRegulatingTerminalFields,
+        setPopupGeneratorEditRegulatingTerminalFields,
+    ] = React.useState(false);
+    const onRegulatingTerminalPopupClose = () => {
+        setPopupGeneratorEditRegulatingTerminalFields(false);
+    };
+    const handleSavePopupGeneratorEditRegulatingTerminalFields = (
+        voltageRegulationGenerator
+    ) => {
+        setGeneratorId(clickedCellData.data.id);
+
+        const isDistantRegulation =
+            voltageRegulationGenerator.voltageRegulationType ===
+            REGULATION_TYPES.DISTANT.id;
+
+        modifyGenerator(
```

Comment:
I think it is a good opportunity to modify this function to make it receive an object instead of all those params

Reply:
we should create a TS to change this function for all equipments